### DISCUSSION TITLE
how to know which part of code need to modify, when change db .

### DISCUSSION THREAD

**Discussion by kin-for-test** [prisma/prisma#19066] [Q&A] [answered]
like Full-text search on PostgreSQL and MySQL

the sql on PostgreSQL  is :
```js
// All drafts that contain the words 'cat' and 'dog'.
const result = await prisma.posts.findMany({
  where: {
    status: 'Draft',
    body: {
      search: 'cat & dog',
    },
  },
})
```

the sql on MYSQL is :
```js
// All drafts that contain the words 'cat' and 'dog'.
const result = await prisma.posts.findMany({
  where: {
    status: 'Draft',
    body: {
      search: '+cat +dog',
    },
  },
})
```

so , when i change my db from PostgreSQL  to MYSQL  (the schema not change ) , how to quickly know which code need to modify?

**Comment by ludralph** [comment, marked as answer]
Hi @kin-for-test 👋 

When migrating from PostgreSQL to MySQL while keeping the same schema, you need to make changes to the code that performs full-text search. In your example, the code snippet for PostgreSQL uses the & operator to perform the search, while the MySQL code snippet uses the + operator.

To quickly identify the parts of your code that need modification, you can search for occurrences of the search property within the where clause. Look for lines similar to:
```js
body: {
  search: 'cat & dog',
},
```
and 
```js
body: {
  search: '+cat +dog',
},
```
These lines indicate the specific code that performs the full-text search. You should modify these lines accordingly based on the syntax requirements of MySQL or PostgreSQL.

Additionally, keep in mind that full-text search capabilities may vary between PostgreSQL and MySQL. While both databases provide full-text search functionality, the specific syntax and features may differ. Therefore, it's essential to consult the documentation to understand the full-text search capabilities and syntax for MySQL or PostgreSQL

**Reply by kin-for-test** [reply]
thank! i understand that ,when i know the project using the full-text search function , then i need to change the & to +,
but how about it is a big project and working with a group together , some one join , some one leave , then how did i know that i need to rewrite witch part of code in the project ? when i change db from one to other , like MySQL to mongodb  , from MySQL to PostgreSQL ,etc

**Reply by ludralph** [reply]
Hi @kin-for-test, There is no easy way, you would need to test the app and check every functionality if you change the database. Usually, there are no modifications required, Full Text Search seems to be an exception because it uses native query syntax (so raw queries would need to be modified as well). Can you tell me more about the use case of switching DB in the middle of a project?
